injection/sharedmain/main.go

```diff
@@ -105,7 +105,13 @@ func GetLoggingConfig(ctx context.Context) (*logging.Config, error) {
 	if err := wait.PollImmediate(1*time.Second, 5*time.Second, func() (bool, error) {
 		var err error
 		loggingConfigMap, err = kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(ctx, logging.ConfigMapName(), metav1.GetOptions{})
-		return err == nil || apierrors.IsNotFound(err), nil
+		if err == nil {
+			return true, nil
+		}
+		if apierrors.IsNotFound(err) || apierrors.IsTimeout(err) || apierrors.IsServerTimeout(err) {
```

Comment:
This makes me a little nervous tbh as I'm not sure if these are the only errors happening on this path.

Could we just store the error and return it eventually like so:

```golang
var lastErr error
if err := wait.PollImmediate(1*time.Second, 5*time.Second, func() (bool, error) {
  loggingConfigMap, lastErr = kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(ctx, logging.ConfigMapName(), metav1.GetOptions{})
  return lastErr == nil || apierrors.IsNotFound(lastErr), nil
}); err != nil {
  return nil, fmt.Errorf("timed out waiting for the condition: %w", lastErr)
}

Reply:
Thanks @markusthoemmes .

I agree with you. I will fix it ASSAP.